### DISCUSSION TITLE
Does @floating-ui/vue support Vue 3?

### DISCUSSION THREAD

**Discussion by alkstal** [floating-ui/floating-ui#2339] [Q&A] [answered]
Trying to install `@floating-ui/vue` in a Vue 3 application and I'm getting a conflicting peer dependency

```
npm ERR! Conflicting peer dependency: vue@2.6.14
npm ERR! node_modules/vue
npm ERR!   peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.1
npm ERR!   node_modules/@vue/composition-api
npm ERR!     peerOptional @vue/composition-api@"^1.0.0-rc.1" from @floating-ui/vue@1.0.0
npm ERR!     node_modules/@floating-ui/vue
npm ERR!       @floating-ui/vue@"*" from the root project
```

Seems like it expects Vue 2 to exist due to an internal dependency?

Is Vue 3 out of the question?

**Comment by atomiks** [comment]
It uses [`vue-demi`](https://github.com/floating-ui/floating-ui/blob/47ee32295b514162571075073d9f3f0e23fcb932/packages/vue/package.json#L68) to support both Vue 2 and Vue 3.  It looks like `vue-demi` has been updated to a new minor (or major in `v0`) and it's probably bringing in an old version as a dependency now. Seems to be problematic that `vue-demi` is still `v0` and can cause this problem, cc @lozinsky 

Although it's probably related to this peerDep: https://github.com/floating-ui/floating-ui/blob/47ee32295b514162571075073d9f3f0e23fcb932/packages/vue/package.json#L58

**Reply by lozinsky** [reply]
> It looks like `vue-demi` has been updated to a new minor (or major in `v0`)

Based on the [release notes](https://github.com/vueuse/vue-demi/releases), it looks like minor.

Yes, we have the `@vue/composition-api` peer dependency but it is marked as optional: https://github.com/floating-ui/floating-ui/blob/47ee32295b514162571075073d9f3f0e23fcb932/packages/vue/package.json#L61-L65

**Comment by lozinsky** [comment]
I tried to install `@floating-ui/vue` with an empty `package.json`:

```json
{
  "dependencies": {}
}
```

And I really got `vue@2.6.14` installed because of `@vue/composition-api`:

```log
40 http fetch GET 200 https://registry.npmjs.org/@floating-ui%2fvue 1417ms (cache revalidated)
41 silly fetch manifest @vue/composition-api@^1.0.0-rc.1
42 http fetch GET 200 https://registry.npmjs.org/@vue%2fcomposition-api 124ms (cache revalidated)
43 silly fetch manifest vue@>= 2.5 < 2.7
44 http fetch GET 200 https://registry.npmjs.org/vue 114ms (cache revalidated)
45 silly placeDep ROOT @floating-ui/vue@1.0.0 OK for:  want: *
46 silly placeDep ROOT vue@2.6.14 OK for: @floating-ui/vue@1.0.0 want: ^2.0.0 || >=3.0.0
47 silly fetch manifest @floating-ui/dom@^1.2.7
48 silly fetch manifest vue-demi@^0.13.11
49 http fetch GET 200 https://registry.npmjs.org/@floating-ui%2fdom 119ms (cache revalidated)
50 http fetch GET 200 https://registry.npmjs.org/vue-demi 240ms (cache revalidated)
51 timing idealTree:#root Completed in 1924ms
```

But when I installed `vue@latest` first:

```json
{
  "dependencies": {
    "vue": "3.3.4"
  }
}
```

And then `@floating-ui/vue`, the dependencies resolved correctly:

```log
41 http fetch GET 200 https://registry.npmjs.org/@floating-ui%2fvue 1255ms (cache revalidated)
42 silly fetch manifest @vue/composition-api@^1.0.0-rc.1
43 http fetch GET 200 https://registry.npmjs.org/@vue%2fcomposition-api 111ms (cache revalidated)
44 silly fetch manifest vue@3.3.4
45 http fetch GET 200 https://registry.npmjs.org/vue 7ms (cache hit)
46 silly fetch manifest vue@>= 2.5 < 2.7
47 silly placeDep ROOT @floating-ui/vue@1.0.0 OK for:  want: *
48 silly fetch manifest @floating-ui/dom@^1.2.7
49 silly fetch manifest vue-demi@^0.13.11
50 http fetch GET 200 https://registry.npmjs.org/@floating-ui%2fdom 110ms (cache revalidated)
51 http fetch GET 200 https://registry.npmjs.org/vue-demi 248ms (cache revalidated)
52 timing idealTree:#root Completed in 1645ms
```

Can you please provide a more verbose error log fragment via `npm i @floating-ui/vue --loglevel verbose` if possible?

**Reply by alkstal** [reply]
I was trying this in a Nuxt 3 application, which doesn't specify any `vue` dependencies.

So that might be why it doesn't work out of the box based on what you tried above.

**Reply by lozinsky** [reply, marked as answer]
Yes, I tried to init clean Nuxt 3 application (via `npx nuxi@latest init my-app`) and got the same error when installing `@floating-ui/vue`. To avoid this you can try installing `@floating-ui/vue` with the `--force` flag, it should work fine with correct Vue version resolution.

I think we can try removing `peerDependencies` and `peerDependenciesMeta` and leave only the `vue-demi` dependency, as done in [`@vueuse/core`](https://github.com/vueuse/vueuse/blob/b7b985b198d9e72136f65fae39b2125cd3abef5b/packages/core/package.json#L41-L46).

**Reply by atomiks** [reply]
@lozinsky are you able to make a PR removing those fields?

**Reply by lozinsky** [reply]
Yes, I will do it today or tomorrow.
